### PR TITLE
Added `attribution` parameter to transactions

### PR DESCRIPTION
**Pull request by ggomaeng**
This fixes errors related to parsing token approvals in some wallets.

Related 
- https://warpcast.com/horsefacts.eth/0xfc6b312e
- https://warpcast.com/undefined/0x054af34f

### CODE REVIEW

File: src/types/transaction.ts [hunk 1]
```diff
@@ -24,9 +24,11 @@ export type ChainIdEip155 = 10 | 8453 | 84532 | 7777777
 export type TransactionParameters = {
   /** A CAIP-2 Chain ID to identify the transaction network. */
   chainId: `${ChainNamespace}:${ChainIdEip155}`
+  /** Optionally omit the client calldata attribution suffix */
+  attribution?: boolean
```
Comment (tmm) [inline src/types/transaction.ts]:
```suggestion
  attribution?: boolean | undefined
```

File: .changeset/five-kids-rush.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"frog": minor
+---
+
+Add `attribution` parameter to transactions
```
Comment (tmm) [inline .changeset/five-kids-rush.md]:
```suggestion
Added `attribution` parameter to transaction response
```

File: src/types/transaction.ts [hunk 3]
```diff
@@ -45,6 +47,8 @@ export type EthSendTransactionParameters<quantity = string> = {
   to: Hex
   /** Value to send with transaction (in wei). */
   value?: quantity
+  /** Optionally omit the client calldata attribution suffix */
+  attribution?: boolean
```
Comment (tmm) [inline src/types/transaction.ts]:
```suggestion
  attribution?: boolean | undefined
```

File: src/types/transaction.ts [hunk 4]
```diff
@@ -79,6 +83,8 @@ export type ContractTransactionParameters<
   abi: abi
   /** Contract function arguments. */
   args?: (abi extends Abi ? UnionWiden<args> : never) | allArgs | undefined
+  /** Optionally omit the client calldata attribution suffix */
+  attribution?: boolean
```
Comment (tmm) [inline src/types/transaction.ts]:
```suggestion
  attribution?: boolean | undefined
```

**Comment by dalechyn** [comment]
I believe it's also worth noting in https://frog.fm/concepts/transactions and set `attribution: false` everywhere with a note that when https://github.com/MetaMask/metamask-extension/pull/23527 is merged, this will be no longer needed.

**Comment by ggomaeng** [comment]
> I believe it's also worth noting in https://frog.fm/concepts/transactions and set `attribution: false` everywhere with a note that when [MetaMask/metamask-extension#23527](https://github.com/MetaMask/metamask-extension/pull/23527) is merged, this will be no longer needed.

That's true. If that's the case, I think it would be better to not merge this PR. 

Instead, it would be better to just provide an example of adding additional specs via middleware like @tmm mentioned [here](https://github.com/wevm/frog/discussions/51#discussioncomment-8732777)

I'll update the PR soon.

**Comment by ggomaeng** [comment]
> I believe it's also worth noting in https://frog.fm/concepts/transactions and set `attribution: false` everywhere with a note that when [MetaMask/metamask-extension#23527](https://github.com/MetaMask/metamask-extension/pull/23527) is merged, this will be no longer needed.

@dalechyn continuing this thread in https://github.com/wevm/frog/pull/177

**Comment by ggomaeng** [comment]
#177 is closed. Continuing in #187

**Pull request closed without merging**